### PR TITLE
Tth htt 2107 take #2 (mini isolation)

### PR DESCRIPTION
**Pull request by ktht**
Note1: This a 3rd attempt b/c I royally screwed up my previous PR since I merged my branch against official remote HEAD 8_0_x branch and blindly pushed to my fork, thus corrupting my development branch.

Note2: Last PR went stale b/c there were conflicts in VHbbAnalysis/Heppy/python/VHbbAnalyzer.py. Added a diff below which shows what the changes would've looked like if the merge would've been successful.

These changes came up in ttH, H->tautau analysis:
- relax lepton pT cuts
  - muons 10 -> 4.5; electrons 10 -> 6.5
  - tested: the number of gen level b-jets vetoed by leptons of lower pT increased by ~1%
  - set electron abs eta cut to 2.5 (default 2.4)
- allow to use different rho for mini-isolation
- include mini-isolation in the isolation cuts as well
- use jet pT value after JEC in jetPtRatio (lepton pT / nearest jet pT)
- modify logic in conditional statements of VhbbAnalyzer so that all the values for Ntuple branches are always calculated whenever the event selection is not applied

Old discussion for the reference: https://github.com/vhbb/cmssw/pull/507, https://github.com/vhbb/cmssw/pull/472

[VHbbAnalyzer.txt](https://github.com/vhbb/cmssw/files/433609/VHbbAnalyzer.txt)

### CODE REVIEW

File: VHbbAnalysis/Heppy/test/vhbb.py [hunk 1]
```diff
@@ -223,12 +223,13 @@
 #	}
 LepAna.mu_isoCorr  = "deltaBeta"
 #LepAna.loose_muon_isoCut = lambda muon : muon.relIso04 < 0.25 
-LepAna.loose_muon_isoCut = lambda muon : muon.relIso04 < 0.4
-LepAna.loose_muon_pt = 10.
+LepAna.loose_muon_isoCut = lambda muon : (muon.relIso04 < 0.4 or muon.miniRelIso < 0.4)
```
Comment (perrozzi) [inline VHbbAnalysis/Heppy/test/vhbb.py:226]: is this general or referred to your analysis only?

**Comment by perrozzi** [comment]
shall we merge?

**Comment by ktht** [comment]
Yes, please merge.

**Comment by arizzi** [comment]
are we sure everybody would be happy with such  change in lepton selection? (Zll? Wlnu?)

**Pull request merged**